### PR TITLE
Allow setting LFS stripe count.

### LINKED ISSUE #6035: Striping the restart.mesh.data file
Hi all.

(Chris Ramos forwarded)

I am running a large 3d model on Frontera with 4000 cpus. A few days later I am notified of this big file, namely the restart.mesh.data.
The problem is all processors are trying to write on this one simultaneously. I take that the situation is more like a traffic jam.

Chris, the Frontera administrator suggests applying the stripping technique on this big restart file. Otherwise, the total number of tasks cannot exceed 840, according to his estimation based on the system capacity.

A further point I learned from Chris is reducing the model size, and therefore reducing the data size won't help much. It's mainly the number of simultaneous checkpointing processes that matters.

I'd like to know whether there is a ready feature in aspect?  Or how to address this issue otherwise?

### PR DESCRIPTION
**Pull request by bangerth**
This is an attempt at making the use of LFS stripe counts a bit easier. Fixes #6035.

@lhy11009 Could you take a look here and potentially give this a try?

### CODE REVIEW

**Comment by lhy11009** [comment]
Fascinating! Thank you for implementing this new feature. Here’s my plan for testing:

I will pull and compile your branch on Frontera, then run the previously problematic case using approximately 4,000 cores with a large checkpoint output file (around 100 GB). I’ll verify that the file striping settings are applied correctly and ensure that the checkpointing is successfully generated.

I’ll keep you updated with the results.

File: source/simulator/parameters.cc [hunk 1]
```diff
@@ -330,7 +330,39 @@ namespace aspect
     prm.declare_entry ("Output directory", "output",
                        Patterns::DirectoryName(),
                        "The name of the directory into which all output files should be "
-                       "placed. This may be an absolute or a relative path.");
+                       "placed. This may be an absolute or a relative path. ASPECT will "
+                       "write output such as statistics files or visualization files "
+                       "into this directory or into directories further nested with in.");
```
Comment (tjhei) [inline source/simulator/parameters.cc:335]:
```suggestion
                       "into this directory or into directories further nested within.");
```
?

File: source/simulator/parameters.cc [hunk 2]
```diff
@@ -433,7 +465,7 @@ namespace aspect
                            Patterns::Double(0., 1.),
                            "A relative tolerance up to which the linear Stokes systems in each "
                            "time or nonlinear step should be solved. The absolute tolerance will "
-                           "then be $\\| M x_0 - F \\| \\cdot \\text{tol}$, where $x_0 = (0,p_0)$ "
+                           "then be $\\| M x_0 - F \\| \\cdot \\text {tol}$, where $x_0 = (0,p_0)$ "
```
Comment (tjhei) [inline source/simulator/parameters.cc:468]: what is going on with these changes?

File: source/simulator/parameters.cc [hunk 3]
```diff
@@ -704,12 +736,12 @@ namespace aspect
     }
     prm.leave_subsection();
 
-    // next declare parameters that pertain to the equations to be
-    // solved, along with boundary conditions etc. note that at this
-    // point we do not know yet which geometry model we will use, so
-    // we do not know which symbolic names will be valid to address individual
-    // parts of the boundary. we can only work around this by allowing any string
-    // to indicate a boundary
+// next declare parameters that pertain to the equations to be
```
Comment (tjhei) [inline source/simulator/parameters.cc:739]: same here?

**Review by tjhei — APPROVED** [review]

**Comment by bangerth** [comment]
Not sure what exactly caused these changes -- I think I accidentally called the indent script when I had un-closed double quotes somewhere. Thanks for pointing it out, it's not fixed.

**Comment by bangerth** [comment]
We have a circularity problem here because I'm introducing a new parameter, reference it in the sphinx documentation (which of course doesn't know anything about it, because it's not in `parameters.md` yet), and then get the error
```
file-system-io.md:11: WARNING: undefined label: 'parameters:output_20directory_20lfs_20stripe_20count' [ref.ref]
```
Should I just run the parameter update script too?

**Comment by gassmoeller** [comment]
Feel free to run the update script for the parameters. We just merged #6098 so it should not change much except for your new parameter in this PR.

**Comment by bangerth** [comment]
OK, this worked.

**Review by tjhei — APPROVED** [review]

**Comment by bangerth** [comment]
@lhy11009 This is now merged. It would be really nice to get feedback on whether it actually works! Thank you in advance!

**Pull request merged**